pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -615,3 +621,39 @@ func GetNodeAddr(node *corev1.Node) (net.IP, error) {
 	}
 	return ipAddr, nil
 }
+
+func (c *Controller) IPInPodSubnets(ip net.IP) bool {
+	var ipCIDR *net.IPNet
+	var curNodeCIDRStr string
+	if ip.To4() != nil {
+		var podIPv4CIDRMaskSize int
+		if c.nodeConfig.PodIPv4CIDR != nil {
+			curNodeCIDRStr = c.nodeConfig.PodIPv4CIDR.String()
+			podIPv4CIDRMaskSize, _ = c.nodeConfig.PodIPv4CIDR.Mask.Size()
+		} else {
+			podIPv4CIDRMaskSize = defaultPodIPv4CIDRMaskSize
+		}
```

Comment:
can you provide some clarity (and maybe add a comment) as to why this is needed? shouldn't we just return `false` at this point?

same for the v6 case below

Reply:
I agree. I thought it might be ok since it is the default setting in Kube-control-manager.
However, thinking more about it it is better to not resolve the flow type and print a warning. This situation also arises for network policy only mode.
https://github.com/vmware-tanzu/antrea/blob/main/pkg/agent/config/node_config.go#L113

Any thoughts @dreamtalen ?

Reply:
I agree too, addressed.